src/operator/tensor/matrix_op.cc

```diff
@@ -23,57 +23,51 @@ DMLC_REGISTER_PARAMETER(ReverseParam);
 
-- ``-2`` for copying all. Pop all remaining input dimensions and push them into
-  the output::
-
-  - input=(2,3,4), shape=(-2), output=(9,8,7)
-  - input=(2,3,4), shape=(2,-2), output=(2,3,4)
-  - input=(2,3,4), shape=(-2,1,1), output=(2,3,4,1,1)
-
-- ``-3`` for merging two dimensions. Pop two input dimensions, compute the product and then
-  push into the output::
-
-  - input=(2,3,4), shape=(-3,4), output=(6,4)
-  - input=(2,3,4), shape=(0,-3), output=(2,12)
-  - input=(2,3,4), shape=(-3,-2), output=(6,4)
-
-- ``-4`` for splitting two dimensions. Pop one input dimensions, next split it
-  according to the next two dimensions (can contain one ``-1``) specified after
-  this code, then push into the output::
-
-  - input=(2,3,4), shape=(-4,1,2,-2), output=(1,2,3,4)
-  - input=(2,3,4), shape=(2,-4,-1,3,-2), output=(2,1,3,4)
+Example::
+   reshape([1,2,3,4], shape=(2,2)) = [[1,2], [3,4]]
 
-If the argument ``reverse`` is set to be true, then translating the input shape
-from right to left. For example, with input shape (10, 5, 4) target shape (-1,
-0), then the output shape will be (50,4) if ``reverse=1``, otherwise it will be
-(40,5).
+Some dimensions of the shape can have special values of 0, -1, -2, -3 or -4 which is interpreted as below: 
+
+- ``0`` copy one dimension from the input to the output shape.
+  Example::
+  - input shape = (2,3,4), shape = (4,0,2), output shape = (4,3,2)
+  - input shape = (2,3,4), shape = (2,0,0), output shape = (2,3,4)
+
+- ``-1`` computes the dimension of the output shape by using the remainder of the input dimensions 
+         to keep the size of the new array same as that of the input array.
+  Example::
+  - input shape = (2,3,4), shape = (6,1,-1), output shape = (6,1,4)
+  - input shape = (2,3,4), shape = (3,-1,8), output shape = (3,1,8)
+  - input shape = (2,3,4), shape=(-1,), output shape = (24,)
+
+- ``-2`` copy all/remainder of the input dimensions to the output shape.
+  Example::
+  - input shape = (2,3,4), shape = (-2,), output shape = (2,3,4)
+  - input shape = (2,3,4), shape = (2,-2), output shape = (2,3,4)
+  - input shape = (2,3,4), shape = (-2,1,1), output shape = (2,3,4,1,1)
+
+- ``-3`` use the product of two consecutive dimensions of the input shape as the output dimension.
+  Example::
+  - input shape = (2,3,4), shape = (-3,4), output shape = (6,4)
+  - input shape = (2,3,4,5), shape = (-3,-3), output shape = (6,20)
+  - input shape = (2,3,4), shape = (0,-3), output shape = (2,12)
+  - input shape = (2,3,4), shape = (-3,-2), output shape = (6,4)
+
+- ``-4 `` split one dimension of the input into two dimensions passed subsequent to -4 in shape(can contain -1).
+  Example::
+  - input shape = (2,3,4), shape = (-4,1,2,-2), output shape =(1,2,3,4)
+  - input shape = (2,3,4), shape = (2,-4,-1,3,-2), output shape = (2,1,3,4)
+
+The argument ``reverse`` if set to 1, then translation of the input shape is done from right to left. 
```

Comment:
"If the argument ``reverse`` is set to 1, then the special values are inferred from right to left. "